Source/WebCore/animation/AnimationTimelinesController.cpp

```diff
@@ -52,32 +52,38 @@ DocumentTimelinesController::DocumentTimelinesController(Document& document)
     }
 }
 
-DocumentTimelinesController::~DocumentTimelinesController() = default;
+AnimationTimelinesController::~AnimationTimelinesController() = default;
 
-void DocumentTimelinesController::addTimeline(DocumentTimeline& timeline)
+void AnimationTimelinesController::addTimeline(AnimationTimeline& timeline)
 {
     m_timelines.add(timeline);
 
+    if (!is<DocumentTimeline>(timeline))
+        return;
+
     if (m_isSuspended)
-        timeline.suspendAnimations();
+        dynamicDowncast<DocumentTimeline>(timeline)->suspendAnimations();
     else
-        timeline.resumeAnimations();
+        dynamicDowncast<DocumentTimeline>(timeline)->resumeAnimations();
 }
 
-void DocumentTimelinesController::removeTimeline(DocumentTimeline& timeline)
+void AnimationTimelinesController::removeTimeline(AnimationTimeline& timeline)
 {
     m_timelines.remove(timeline);
 }
 
-void DocumentTimelinesController::detachFromDocument()
+void AnimationTimelinesController::detachFromDocument()
 {
     m_currentTimeClearingTaskCancellationGroup.cancel();
 
-    while (!m_timelines.isEmptyIgnoringNullReferences())
-        m_timelines.begin()->detachFromDocument();
+    while (!m_timelines.isEmptyIgnoringNullReferences()) {
+        auto& timeline = *m_timelines.begin();
+        if (is<DocumentTimeline>(timeline))
+            dynamicDowncast<DocumentTimeline>(timeline)->detachFromDocument();
```

Comment:
```suggestion
        if (RefPtr documentTimeline = dynamicDowncast<DocumentTimeline>(timeline))
            documentTimeline->detachFromDocument();
```

This is the more common pattern to avoid checking the pointer twice

Reply:
Fixed.